src/routing/mod.rs

```diff
@@ -357,24 +357,38 @@ impl Routing {
             public_key,
         }) = itinerary.dst
         {
-            let socket_addr = self
-                .dispatcher
-                .core
-                .lock()
-                .await
-                .get_socket_addr(socket_id)
-                .copied();
-
-            if let Some(socket_addr) = socket_addr {
-                return self
-                    .send_message_to_client(socket_addr, ClientMessage::from(content)?)
-                    .await;
+            let name = if let Some(pub_key) = public_key.ed25519() {
```

Comment:
```suggestion
            let name =XorName::from(public_key)
```
As the client can have any type of Keypair(BLS/Ed). And we do have a `From` implemented for `sn_data_types::PublicKey` [here](https://github.com/maidsafe/sn_data_types/blob/095e22eb36b381a01397bb7708878d0d56582974/src/keys/public_key.rs#L177).